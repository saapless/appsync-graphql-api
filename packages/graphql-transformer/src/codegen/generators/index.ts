export { DynamoDbGenerator } from "./DynamoDBGenerator";
export { TypeGenerator } from "./type";
