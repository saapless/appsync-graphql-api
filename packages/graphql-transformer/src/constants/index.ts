export const ConnectionDirective = Object.freeze({
  HAS_ONE: "hasOne",
  HAS_MANY: "hasMany",
});

export const ModelDirective = Object.freeze({
  MODEL: "model",
});

export const UtilityDirective = Object.freeze({
  SERVER_ONLY: "serverOnly",
  CLIENT_ONLY: "clientOnly",
  READ_ONLY: "readOnly",
  WRITE_ONLY: "writeOnly",
  FILTER_ONLY: "filterOnly",
  INTERNAL: "internal",
});

export const ScalarType = Object.freeze({
  STRING: "String",
  INT: "Int",
  FLOAT: "Float",
  BOOLEAN: "Boolean",
  ID: "ID",
  LONG: "Long",
});

export const RelationType = Object.freeze({
  ONE_TO_ONE: "oneToOne",
  ONE_TO_MANY: "oneToMany",
  MANY_TO_MANY: "manyToMany",
});
