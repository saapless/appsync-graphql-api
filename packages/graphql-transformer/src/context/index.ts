export * from "./TransformerContext";
export type {
  DescriptorOperation,
  Key,
  KeyOperator,
  KeyValue,
  OperationActionType,
  ResolverDescriptor,
  ReturnType,
} from "./ResolverLoader";
