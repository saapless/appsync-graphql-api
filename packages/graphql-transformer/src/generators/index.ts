export { ResolverGeneratorBase } from "./ResolverGeneratorBase";
export { DynamoDbGenerator } from "./DynamoDBGenerator";
export { NoneGenerator } from "./NoneGenerator";
export { SchemaTypesGenerator } from "./SchemaTypesGenerator";
export { ResolverTypesGenerator } from "./ResolverTypesGenerator";
