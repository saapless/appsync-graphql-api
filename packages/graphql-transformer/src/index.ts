export * from "./transformer";
export { definitionFromSource } from "./utils";
