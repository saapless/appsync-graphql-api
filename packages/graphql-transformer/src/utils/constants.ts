/**
 * DIRECTIVES
 */
export const AUTH_DIRECTIVE = "auth";

export const AWS_AUTH_DIRECTIVE = "aws_auth";
export const AWS_API_KEY_DIRECTIVE = "aws_api_key";
export const AWS_IAM_DIRECTIVE = "aws_iam";
export const AWS_OIDC_DIRECTIVE = "aws_oidc";
export const AWS_LAMBDA_DIRECTIVE = "aws_lambda";
export const AWS_COGNITO_USER_POOLS_DIRECTIVE = "aws_cognito_user_pools";
export const AWS_SUBSCRIBE_DIRECTIVE = "aws_subscribe";

export const HAS_ONE_DIRECTIVE = "hasOne";
export const HAS_MANY_DIRECTIVE = "hasMany";

export const DATA_SOURCE_DIRECTIVE = "dataSource";
export const RESOLVER_DIRECTIVE = "resolver";

export const MODEL_DIRECTIVE = "model";

export const SERVER_ONLY_DIRECTIVE = "serverOnly";
export const CLIENT_ONLY_DIRECTIVE = "clientOnly";
export const READ_ONLY_DIRECTIVE = "readOnly";
export const WRITE_ONLY_DIRECTIVE = "writeOnly";
export const FILTER_ONLY_DIRECTIVE = "filterOnly";

export const DEFAULT_ID_FIELD = "id";
export const DEFAULT_SK_FIELD = "_sk";

export const DEFAULT_SOURCE_INDEX = "bySourceId";
export const DEFAULT_SOURCE_INDEX_FIELD = "sourceId";
export const DEFAULT_TARGET_INDEX = "byTargetId";
export const DEFAULT_TARGET_INDEX_FIELD = "targetId";
export const DEFAULT_TYPENAME_INDEX = "byTypename";
export const DEFAULT_TYPENAME_INDEX_FIELD = "__typename";
