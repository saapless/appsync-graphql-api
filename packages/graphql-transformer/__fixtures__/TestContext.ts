import { TransformerContext, TransformerContextConfig } from "../src/context";

interface TestingTransformerConfig extends TransformerContextConfig {
  allowPrint?: boolean;
}

export class TestContext extends TransformerContext {
  public allowPrint: boolean;
  public files: Map<string, string> = new Map();

  constructor(config: TestingTransformerConfig) {
    super(config);
    this.allowPrint = config.allowPrint ?? false;
    this.files = new Map();
  }

  public override createOutputDirectory(path: string) {
    return path;
  }

  public printFile(filePath: string, content: string) {
    if (this.allowPrint) {
      return super.printFile(filePath, content);
    }

    this.files.set(filePath, content);
  }

  public printScript(filePath: string, content: string) {
    if (this.allowPrint) {
      return super.printScript(filePath, content);
    }

    this.files.set(filePath, content);
  }
}
