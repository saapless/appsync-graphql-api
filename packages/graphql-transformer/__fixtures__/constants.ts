import { DataSourceManagerConfig } from "../src/context/DataSourceManager";

export const TEST_DS_CONFIG = {
  primaryDataSourceName: "TestDataSource",
  dataSources: {
    TestDataSource: {
      type: "DYNAMO_DB",
      config: {
        tableName: "TestTable",
      },
    },
    NoneDataSource: {
      type: "NONE",
    },
  },
} satisfies DataSourceManagerConfig;
