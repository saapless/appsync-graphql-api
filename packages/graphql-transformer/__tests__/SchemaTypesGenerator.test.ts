import { TEST_DS_CONFIG } from "../__fixtures__/constants";
import { TransformerContext } from "../src/context";
import { DocumentNode } from "../src/definition";
import { SchemaTypesGenerator } from "../src/generators";

const context = new TransformerContext({
  document: DocumentNode.fromSource(/* GraphQL */ `
    type User {
      id: ID!
      name: String!
    }

    input CreateUserInput {
      id: ID
      name: String
    }

    input UpdateUserInput {
      id: ID!
      name: String
    }

    type Query {
      me: User
    }

    type Mutation {
      createUser(input: CreateUserInput!): User
      updateUser(input: UpdateUserInput!): User
      deleteUser(id: ID!): User
    }
  `),
  dataSourceConfig: TEST_DS_CONFIG,
});

describe("SchemaTypesGenerator", () => {
  it("genrates schema types", () => {
    const generator = new SchemaTypesGenerator(context);

    const types = generator.generate("schema-types.ts");
    expect(types).toMatchSnapshot();
  });
});
