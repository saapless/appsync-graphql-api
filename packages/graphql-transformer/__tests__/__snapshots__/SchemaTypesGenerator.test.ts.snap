// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaTypesGenerator genrates schema types 1`] = `
"export type Maybe<T> = T | null | undefined;
export type User = {
    id: string;
    name: string;
};
export type CreateUserInput = {
    id?: Maybe<string>;
    name?: Maybe<string>;
};
export type UpdateUserInput = {
    id: string;
    name?: Maybe<string>;
};
export type Query = {
    me?: Maybe<User>;
};
export type MutationCreateUserArgs = {
    input: CreateUserInput;
};
export type MutationUpdateUserArgs = {
    input: UpdateUserInput;
};
export type MutationDeleteUserArgs = {
    id: string;
};
export type Mutation = {
    createUser?: Maybe<User>;
    updateUser?: Maybe<User>;
    deleteUser?: Maybe<User>;
};
"
`;
